rust/src/delta.rs

```diff
@@ -540,8 +550,35 @@ impl DeltaTable {
             self.version(),
         );
 
-        while let PeekCommit::New(new_version, actions) =
-            self.peek_next_commit(self.version()).await?
+        // construct stream yielding (version, bytes)
+        let store = self.storage.clone();
+        let log_stream = futures::stream::iter(self.version() + 1..).map(|i| {
+            let store2 = store.clone();
+            let loc = commit_uri_from_version(i).clone();
+            async move {
+                let out = store2.get(&loc);
+                (i, out.await)
+            }
+        });
+
+        let buf_size = 50; // TODO
+
+        // why is mut needed here and is it OK?
+        let mut log_buffer = {
+            match max_version {
+                Some(n) => log_stream.take((n - self.version() + 2).try_into().unwrap()).buffered(buf_size),
+                None => log_stream.take(usize::MAX).buffered(buf_size) // need take to have compatible types
+            }
+        };
+        
+        while let Some((new_version, actions)) = {
+            let next_commit = log_buffer.next().await.unwrap();
```

Comment:
```suggestion
            let next_commit = log_buffer.next().await?;
```

This might be warranted as well, unwraps will panic the caller on errors which is not something we want

Reply:
In this case the `await` returned an `Option` so I just handled it as such in the match below. Now the case when the stream ends before expected is explicitly an error